internal/sourceanalysis/go.go

```diff
@@ -25,6 +25,15 @@ func goAnalysis(r reporter.Reporter, pkgs []models.PackageVulns, source models.S
 		return
 	}
 
+	// Set GOVERSION to the Go version in go.mod.
+	var goVersion string
+	for _, pkg := range pkgs {
+		if pkg.Package.Name == "stdlib" {
+			goVersion = pkg.Package.Version
+		}
+	}
+	os.Setenv("GOVERSION", fmt.Sprintf("go%s", goVersion))
+
 	vulns, vulnsByID := vulnsFromAllPkgs(pkgs)
 	res, err := runGovulncheck(filepath.Dir(source.Path), vulns)
```

Comment:
Can the Go version be passed down to `runGovulncheck` instead? `govulncheck` supports adding env variables to the scan command. https://pkg.go.dev/golang.org/x/vuln/scan#Cmd

`os.Setenv` affects the whole process, so we may get unexpected results when run concurrently. The first thing that comes to mind is tests, either here in osv-scanner, or Scorecard. Some of our e2e tests call osv-scanner concurrently from different Go routines.

Reply:
+1

Reply:
Thanks! updated code